---
title: "RNAseq_230214"
author: "Andrea"
date: '2023-03-01'
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
```


Project directory: /scratch/teams/dawson_genomics/Projects/DNMTi/RNAseq/230214_NB501056_0923_AH3JFCBGXT

fastq dir: /pipeline/Runs/NextSeq/230214_NB501056_0923_AH3JFCBGXT/ProjectFolders/Project_Brian-Liddicoat/Sample_*/

```{bash}
# set up project dir
cd /scratch/teams/dawson_genomics/Projects/DNMTi/RNAseq/

mkdir 230214_NB501056_0923_AH3JFCBGXT
cd 230214_NB501056_0923_AH3JFCBGXT

mkdir fastqc
mkdir scripts
mkdir logs

for fq in /pipeline/Runs/NextSeq/230214_NB501056_0923_AH3JFCBGXT/ProjectFolders/Project_Brian-Liddicoat/Sample_*/*_R1_001.fastq.gz
do
sbatch scripts/fastqc.sbatch $fq 
done

# when fastqc is finished run multiqc 
module load multiqc/1.8
cd fastqc
multiqc .
cd ..
```

# align using ngm slam-seq settings as done for previous RNAseq
```{bash}
# run from project dir
for fq in /pipeline/Runs/NextSeq/230214_NB501056_0923_AH3JFCBGXT/ProjectFolders/Project_Brian-Liddicoat/Sample_*/*_R1_001.fastq.gz
do
sbatch scripts/ngm_align_paired_v043.sbatch $fq 
done

cd trimmed_fastqs
multiqc .
cd ..
```

# make TDFs
```{bash}
for bam in bams/*_sort_ngmRG_filtered_dedup.bam
do
sbatch scripts/toTDF.sbatch $bam
done
```

# use htseq for counts
```{bash}
for bam in bams/*_sort_ngmRG_filtered_dedup.bam
do
sbatch scripts/htseq_count.sbatch $bam
done
```

# set up dirs
```{r}
project.dir <- "/scratch/teams/dawson_genomics/Projects/DNMTi/RNAseq/230214_NB501056_0923_AH3JFCBGXT"
counts.dir <- file.path(project.dir, "counts")
session.dir <- file.path(project.dir, "session-info")
bam.dir <- file.path(project.dir, "bams")

dir.create(session.dir)
```

# load into edger and make sample ann
```{r}
library(edgeR)

# get a list of all TCcount files
files <- list.files(counts.dir)
# simplify names to use as sample names
snames <- gsub("DNMT1i-RNA-MV411-dC9-", "", gsub(".count", "", files))

# make sample annotation 
sample.ann <- data.frame(
  "files" = files, 
  "samples" = snames, 
  "cells" = rep(c(rep("4KD", 3), rep("4NT", 3)), 2),
  "treatment" = c(rep("DMSO", 6), rep("DNMTi", 6)),
  "replicate" = rep(paste0("R", seq(1, 3, 1)), 4)
  )
sample.ann$group <- paste(sample.ann$cells, sample.ann$treatment, sep = "_")

dge.counts = readDGE(
  file = sample.ann$files, 
  path = counts.dir,
  group = sample.ann$group
  )
```

# Filtering out genes with low counts & not useful
```{r}
noint <- rownames(dge.counts$counts) %in% c("__no_feature", "__ambiguous", "__too_low_aQual","__not_aligned","__alignment_not_unique")
table(noint)
counts.cpms <- cpm(dge.counts$counts)
keep <- rowSums(counts.cpms > 1 ) >= 2 & !noint
dim(dge.counts$counts)
dge.counts$counts <- dge.counts$counts[keep,]
dim(dge.counts$counts)
```
> dim(dge.counts$counts)
[1] 43392    12
#dge.counts$counts <- dge.counts$counts[keep,]
> dim(dge.counts$counts)
[1] 14103    12

# calc TMM norm and write out file
```{r}
results.dir <- file.path(project.dir, "results")

dge.counts <- calcNormFactors(dge.counts, method="TMM")

# multiply counts by norm factors
counts.norm <- dge.counts$counts%*%diag(dge.counts$samples$norm.factors)
colnames(counts.norm) <- snames
write.table(
  counts.norm,
  file.path(results.dir, "4KD_TMMnorm_counts.txt"),
  sep = "\t",
  col.names = NA,
  row.names = TRUE,
  quote = FALSE
)
```

# add ann info
```{r}
dge.counts$samples$treatment = sample.ann$treatment
dge.counts$samples$cells = sample.ann$cells
dge.counts$samples$replicate = sample.ann$replicate
```

# MDS plot norm counts
```{r}
library(RColorBrewer)

plots.dir <- file.path(project.dir, "plots")

# Setting colours
col.group <- as.factor(dge.counts$samples$group)
levels(col.group) <- brewer.pal(nlevels(col.group), "Set1") 
col.group <- as.character(col.group)

pdf(file.path(plots.dir, "MDS_TMM_norm_counts.pdf"))
MDS = plotMDS(
  dge.counts,
  labels = paste0(dge.counts$samples$group, "_", dge.counts$samples$replicate), 
  col = col.group
  )
dev.off()
```

# set design and contrasts for DE
```{r}
# design table
design = model.matrix(~0+group, data = dge.counts$samples)
design

contr.matrix <- makeContrasts(
  DNMTivsDMSO_4NT = group4NT_DNMTi-group4NT_DMSO,
  DNMTivsDMSO_4KD = group4KD_DNMTi-group4KD_DMSO,
  levels = colnames(design)
  )
contr.matrix
```

# estimate dispersion and get DE
```{r}
dge.counts = estimateDisp(dge.counts, design)

gfit <- glmQLFit(dge.counts, design)
ftest.list <- list()

for(contrast.i in 1:length(colnames(contr.matrix))){
  
  ftest.list[[contrast.i]] <- glmQLFTest(gfit, contrast = contr.matrix[ , contrast.i])
  # write ftest table
  write.table(
    ftest.list[[contrast.i]]$table,
    file.path(results.dir, paste0(colnames(contr.matrix)[contrast.i], "_ftest_table.txt")),
    sep = "\t",
    quote = F,
    col.names = T,
    row.names= F
  )
}
```

# make heatmap of genes sig up w/drug in WT cells
```{r}
library(ComplexHeatmap)
drug.up <- subset(ftest.list[[1]]$table, PValue < 0.05 & logFC > 0)

# get normalised counts for genes in drup.up table
heat.counts <- dge.counts$counts[which(rownames(dge.counts$counts) %in% rownames(drug.up)), ] %*% diag(dge.counts$samples$norm.factors)
colnames(heat.counts) <- snames

heat.mat <- as.matrix(heat.counts)

pdf(file.path(plots.dir, "Heatmap_TMMnorm_expression_4NT_drug_up.pdf"))
Heatmap(heat.mat, cluster_rows = TRUE, cluster_columns = FALSE, show_row_names = TRUE, use_raster = FALSE)
dev.off()

# limit to genes with higher LFC in WT drugvDMSO
drug.up <- subset(ftest.list[[1]]$table, PValue < 0.05 & logFC > 1)

# get normalised counts for genes in drug.up table
heat.counts <- dge.counts$counts[which(rownames(dge.counts$counts) %in% rownames(drug.up)), ] %*% diag(dge.counts$samples$norm.factors)
colnames(heat.counts) <- snames

# also require higher coverage
keep <- rowSums(heat.counts > 100) >= 4
heat.counts <- heat.counts[keep,]
heat.mat <- as.matrix(log10(heat.counts))
# change -INF values (where initial value was 0) to 0
heat.mat[is.infinite(heat.mat)] <- 0

col_fun = rev(brewer.pal(11, name = "Spectral"))
pdf(file.path(plots.dir, "Heatmap_TMMnorm_LOGexpression_100_4NT_drug_up_LFC1.pdf"))
Heatmap(heat.mat, cluster_rows = TRUE, cluster_columns = FALSE, show_row_names = TRUE, use_raster = FALSE, col = col_fun, row_names_gp = gpar(fontsize = 1), column_names_gp = gpar(fontsize = 6), name = "log10(TMM)")
dev.off()
```

# re-align with STAR to match previous RNAseq analysis, get counts and tdfs also
```{bash}
for fq in /pipeline/Runs/NextSeq/230214_NB501056_0923_AH3JFCBGXT/ProjectFolders/Project_Brian-Liddicoat/Sample_*/*_R1_001.fastq.gz
do
sbatch scripts/align_star.sbatch $fq 
done

for bam in STAR_bams/*_Aligned.out.bam
do
sbatch scripts/STAR_htseq_count.sbatch $bam 
done
# needs to be re-run for DNMT1i-RNA-MV411-dC9-0nM-4NT-R3_S9_Aligned.out.bam with 8 hr time limit (4hr is fine for others)

for bam in STAR_bams/*_Aligned.out.bam
do
sbatch scripts/STAR_toTDF.sbatch $bam
done
# needs to be re-run for DNMT1i-RNA-MV411-dC9-0nM-4NT-R3_S9_Aligned.out.bam with 6 hr time limit (2hr is fine for others)
```

# load into edger and get norm counts for heatmap
```{r}
star.counts <- file.path(project.dir, "STAR_counts")
star.results <- file.path(project.dir, "STAR_results")

# get a list of all TCcount files
files <- list.files(star.counts)
# simplify names to use as sample names
snames <- gsub("DNMT1i-RNA-MV411-dC9-", "", gsub(".count", "", files))

dge.counts = readDGE(
  file = sample.ann$files, 
  path = star.counts,
  group = sample.ann$group
  )

# write out raw counts
write.table(
  dge.counts$counts,
  file.path(star.results, "4KD_raw_counts.txt"),
  sep = "\t",
  row.names = TRUE,
  col.names = NA,
  quote = FALSE
)

# Filtering out genes with low counts & not useful
noint <- rownames(dge.counts$counts) %in% c("__no_feature", "__ambiguous", "__too_low_aQual","__not_aligned","__alignment_not_unique")
table(noint)
counts.cpms <- cpm(dge.counts$counts)
keep <- rowSums(counts.cpms > 1 ) >= 2 & !noint
dim(dge.counts$counts)
dge.counts$counts <- dge.counts$counts[keep,]
dim(dge.counts$counts)
```
> dim(dge.counts$counts)
[1] 43392    12
#dge.counts$counts <- dge.counts$counts[keep,]
> dim(dge.counts$counts)
[1] 13800    12

# calc TMM norm and write out file
```{r}
dge.counts <- calcNormFactors(dge.counts, method = "TMM")

# multiply counts by norm factors
counts.norm <- as.data.frame(dge.counts$counts%*%diag(dge.counts$samples$norm.factors))
colnames(counts.norm) <- snames
write.table(
  counts.norm,
  file.path(star.results, "4KD_TMMnorm_counts.txt"),
  sep = "\t",
  col.names = NA,
  row.names = TRUE,
  quote = FALSE
)
```

# get IFN response genes MSigDB pathway (same as Naiara used) to make heatmap in 4KD samples
```{r}
library(data.table)
library(RColorBrewer)

# get MSigDB rds from Naiara's dir
Hs_H.all <- readRDS(file="/dawson_genomics/Projects/DNMTi/RNAseq/210108_NB501056_0582_AHMHNFAFX2/Hs.h.all.v7.1.entrez.rds")

# get IFN genes, this uses entrez IDs
ifn.entrez <- Hs_H.all[["HALLMARK_INTERFERON_GAMMA_RESPONSE"]]

# annotate entrez ids on to norm counts
entrez.sym <- fread("/data/reference/dawson_labs/biomart_annotations/Hsapiens/ensembl_hgnc_entrez.txt")
counts.norm$hgnc <- rownames(counts.norm)
counts.norm <- merge(counts.norm, entrez.sym, by.x = "hgnc", by.y = "hgnc_symbol")

# limit to ifn genes for hm, and format appropriately
hm.counts <- counts.norm[which(counts.norm$entrezgene_id %in% ifn.entrez),]
hm.counts <- hm.counts[-which(duplicated(hm.counts$hgnc)),]
hm.mat <- as.matrix(log10(hm.counts[, 2:13]))
rownames(hm.mat) <- hm.counts$hgnc
# change -INF values (where initial value was 0) to 0
hm.mat[is.infinite(hm.mat)] <- 0

pdf(file.path(project.dir, "STAR_plots/Heatmap_TMMnorm_LOGexpression_MSigDB_IFN_genes.pdf"))
Heatmap(hm.mat[, -c(5:6)], cluster_rows = TRUE, cluster_columns = FALSE, show_row_names = TRUE, use_raster = FALSE, col = col_fun, row_names_gp = gpar(fontsize = 3), column_names_gp = gpar(fontsize = 6), name = "log10(TMM)")
dev.off()

# remove poorly sequences samples and zscore heatmap
zhm.mat <- t(scale(t(hm.mat[, -c(5:6)])))
pdf(file.path(project.dir, "STAR_plots/Heatmap_zscore_LOGexpression_MSigDB_IFN_genes.pdf"))
Heatmap(zhm.mat, cluster_rows = TRUE, cluster_columns = FALSE, show_row_names = TRUE, use_raster = FALSE, col = col_fun, row_names_gp = gpar(fontsize = 3), column_names_gp = gpar(fontsize = 6), name = "log10(TMM)")
dev.off()
```

# due to quality issues in MGC (see multiQC), these are being re-sequenced
# run fastqc on 2 sample shallow sequencing of repeat samples (run interactive on prod_short)
```{bash}
module load fastqc/0.11.6
module load multiqc/1.8

mkdir shallow_fastqc

fastqc -o shallow_fastqc/ /pipeline/Runs/NextSeq/230412_NB501056_0944_AHK5YYBGXT/ProjectFolders/Project_Stuart-Craig/Sample_DNMT1i-RNA-MV411-dC9-0nM-4NT-R3/*fastq.gz
fastqc -o shallow_fastqc/ /pipeline/Runs/NextSeq/230412_NB501056_0944_AHK5YYBGXT/ProjectFolders/Project_Stuart-Craig/Sample_DNMT1i-RNA-MV411-dC9-100nM-4KD-R2/*fastq.gz

cd shallow_fastqc
multiqc .
cd ..
```

# fastq screen now on cluster, run original samples through to check for contamination and add to multiqc
```{bash}
for fq in /pipeline/Runs/NextSeq/230214_NB501056_0923_AH3JFCBGXT/ProjectFolders/Project_Brian-Liddicoat/Sample_*/*_R1_001.fastq.gz
do
sbatch scripts/fastq_screen.sbatch $fq 
done

module load multiqc/1.8
multiqc -o multiqc .
```



# save session info
```{r}
sesh.info <- capture.output(sessionInfo())
writeLines(
  sesh.info,
  file.path(session.dir, paste0(Sys.Date(), "_RNAseq_230214.txt"))
)
```
